src/main/java/javaserver/handlers/FileHandler.java

```diff
@@ -1,67 +1,54 @@
 package javaserver.handlers;
 
 import javaserver.Request;
+import javaserver.utility.ResourceUtility;
 
 import java.util.Arrays;
-import java.io.File;
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.util.List;
 import java.io.IOException;
 
 public class FileHandler implements Handler {
 
-    private String fileUri;
-    private Response response = new Response();
+    private final static List ACCEPTABLE_IMAGE_FORMATS = Arrays.asList(".jpg", ".gif", ".png");
+
     private int contentSize = 0;
+    protected ResourceUtility resourceUtility;
+    protected String fileUri;
+    protected Response response = new Response();
 
-    public FileHandler(String fileUri) {
+    public FileHandler(String fileUri, ResourceUtility resourceUtility) {
         this.fileUri = fileUri;
+        this.resourceUtility = resourceUtility;
     }
 
     public Response handleRequest(Request request) {
-        response = new Response();
 
         String method = request.getMethod();
         if (!method.equals("GET")) {
             response.setStatus(Status.MethodNotAllowed);
             return response;
         }
 
-        byte[] body = getFileContents(fileUri);
-        if (request.getHeaders().containsKey("Range")) {
-            String range = request.getHeaders().get("Range");
-            int start = getFirstNumberInRange(range, body.length);
-            int end = getLastNumberInRange(range, body.length);
-
-            body = getRangeOfBytes(body, start, end);
-
-            String headers = "";
-            headers = "Content-Length: " + body.length + "\r\n";
-            headers += "Content-Type: txt/plain" + "\r\n";
-            response.setHeaders(headers);
-
-            response.setStatus(Status.PartialContent);
-        } else {
-            String uri = request.getUri();
-            String[] splitUri = uri.split("\\.");
-
-            if (splitUri.length > 1) {
-                String headers = "";
-                headers = "Content-Length: " + contentSize + "\r\n";
-                headers += "Content-Type: image/" + splitUri[1] + "\r\n";
-                response.setHeaders(headers);
-            }
+        if (isPartialRequest(request)) {
+            response = new PartialHandler(fileUri, resourceUtility).handleRequest(request);
+            return response;
```

Comment:
I think you can just return these directly - instead of setting `response` and then returning it